### ISSUE TITLE
How to SAVE the final PDF to a specific location with a user defined name in the PDF Preview Pane

### ISSUE THREAD

**Issue by davidc86** [sillsdev/ptx2pdf#1014]
I don't see any way to SAVE the final PDF to a specific location with a user defined name from the PDF Preview Pane.

When we used Sumatra PDF Reader, there was a standard File menu. This allowed us to save the final, beautifully finished PDF file to a user-defined location with a user-specified name. This helps with file management and keeps the "final" versions safe from being accidentally overwritten in the future. 

This is not a "big deal", but it seems a "Save..." button would be a reasonable addition at some point, as it makes it much easier for the end-user to keep organized. 

"A User Request"

**Comment by markpenny** [comment]
Okay, I like the idea. And this comes on the heels of someone yesterday sending us their problematic PTXprint-generated PDF which had then been "printed" by Microsoft's PDF writer to make a PDF (in order to send it to us). Scream!! :-) 

I am thinking we should have a few different options available on the Preview Pane:

- Save As...
- Open Folder (where the PDF is located)
- Send PDF by e-mail
- Copy Filename (Path) to clipboard
- Print... (we already have this - and it is rather limited at present)
- Show report - something like PDF Properties, but with a lot more useful information such as Fonts used, Config name, PDF type, etc.

What else do we need to include? It probably won't happen until next year, but this is a good time to ask Santa for what you want NEXT Christmas :-)

**Comment by davidc86** [comment]
Hahaha! Yup we'll ask away, maybe we'll get it.

I like these ideas.

Most useful to me is: Save As...
I also like Open Folder

I think many might prefer "Share PDF..." rather than "Send PDF by email' because at least in Indonesia almost everybody now sends stuff to each other via WhatsApp. So, we click "Share PDF...", then select WhatsApp, then pick the target person and off it goes. (And the Share menu also includes the option to email it.)

I doubt many of our teams would use "Copy Filename (path) to clipboard" because that requires another layer of abstraction to know what to do with that information.

I really like the "Show report on PDF" or something similar to "Document Properties" that Acrobat Reader provides. Is that what you are thinking here? Maybe just call it "PDF properties"? Unless you are wanting to give more information that just what's in the PDF file.

You say there is a Print... button. Is there? I have not found it yet, unless you mean the Print (Make PDF) button on the main screen. But that label is a bit confusing. Maybe "Create PDF" would be a clearer label for that button? (Or just "Make PDF"), because in other apps, the Print... menu option calls the print dialog, and gives choices like which printer are we going for, etc. Currently, PTXprint doesn't print anything; it creates a PDF file. (Very nicely, btw.) The missing bit is how to get to that file so we can then share, email, upload, rename, copy, move, delete, or print it.

Adding a simple Print... option that calls the print dialog would be very helpful too. (Whether or not you change the name of the main button on PTXprint!)


________________________________

**Comment by markpenny** [comment]
It just goes to show that if you don't use standard icons, then no one actually reads the labels on the buttons :-) 
![Image](https://github.com/user-attachments/assets/b0356ac2-79dd-42c5-b146-b9c250d0f89d)

**Comment by davidc86** [comment]
HAHAHAHAHA!
DUDE!
I totally looked at that button several times yesterday and it didn't really register that it was a simple Print... button. Though I do recall thinking at one point, "Oh that's nice they've added the ability to put it on paper directly." But I was so busy trying to find out how to Save the PDF to a place I wanted with a name I wanted that the Print (on Paper) button never fully stuck in my brain!

Besides everyone knows a print option is Print... or an icon of a printer, so clearly THAT button can't be that, can it?

(Maybe I actually am getting old.)
________________________________

**Comment by markpenny** [comment]
Okay, I've just released 2.7.7 with a few more OBVIOUS buttons :-)

![Image](https://github.com/user-attachments/assets/5896d851-4cc2-4ba7-bd02-1c18f7faf80c)

Haven't yet worked out how to code up the Share... button, but hopefully that will happen soon. I am also hoping that placing them at the top-left might make them more discoverable too.

**Comment by davidc86** [comment]
Nice. The Save a COPY button works just like we needed. That was fast!

One thing: please check on your computer if your icons look "live". Because on mine, the "Open Folder" and the "Print" icon both look active and work as advertised. But the two middle ones: "Save a Copy" and "Share" both **_look inactive_** -- even though the "Save a Copy" button actually works. So that's a bit confusing.

**Comment by markpenny** [comment]
Yes, I've now seen this on another machine, and will try to find out why it isn't correct.
This is what it should look like, but the 2nd icon is definitely _different_ and also gray/_faint_:
![Image](https://github.com/user-attachments/assets/702b7910-6620-4f04-a1e9-a204815ae7b3)

**Comment by davidc86** [comment]
BTW, is there ANY way for PTXprint to remember the folder we navigated to and saved the PDF into? 

Like maybe it could be written it into the project somewhere? 

This would save us a lot of time since the file-manager PTXprint uses does not recognize Windows shortcuts in the PTXprint folder as valid pointers to folders we go to frequently. And the folder we use is embedded like 7 layers in. 

Again, not a "biggie" but certainly a nice enhancement.

**Comment by markpenny** [comment]
Yes, we can do that. I would have done so yesterday, but was in a hurry.
Mark

**Comment by davidc86** [comment]
Woohoo! You fixed the Save as icon and you made it so it remembers where we saved the file the last time (even after restarting PTXprint!) Thanks for that. Very helpful.

I won't CLOSE this in case, Mark, you want the early discussion on the Share button. But if you want to close this and create a new "Share" button "enhancement", feel free.
